### DISCUSSION TITLE
Memory occupied for a long time while running passthrough but not released

### DISCUSSION THREAD

**Discussion by Roay** [libfuse/libfuse#977] [Q&A]
kernel version：3.10
libfuse version：3.11
mount command   ：./passthrough /mnt
tree command line ：tree /mnt/var/8000w_dir

Test instructions:
1. There are 8kw files in the directory "/var/8000kw_dir" (created through vdbench)
2. During the execution of the tree, the maximum memory occupied by the process ‘passthrough’ is approximately 16.7GB.
3. After the tree execution is completed, the process ‘passthrough’ occupies approximately 13.3G of memory.
4. After executing "echo 3>/proc/sys/vm/drop_caches", the process ‘passthrough’ occupies approximately 10GB of memory.

I want to know:
1. Which structure occupies so much memory space, is it dentry&inode?
2. Why "echo 3>/proc/sys/vm/drop_caches"  cannot be released?
3. How can memory be automatically released to avoid OOM situations?

**Comment by bsbernd** [comment]
Unless you show that there is a real issue with current libfuse version, this is not an issue but a request for help - discussion by definition. We can *assist* you to find out the issue, but finding leaks is something you can do yourself (LSAN,valgrind, jemalloc...)
